ibis/expr/types/generic.py

```diff
@@ -189,33 +192,53 @@ def null():
 
 
 @public
-def literal(value, type=None):
+def literal(value: Any, type: dt.DataType | str | None = None) -> ScalarExpr:
     """Create a scalar expression from a Python value.
 
+    !!! tip "Use specific functions for arrays, structs and maps"
+
+        Ibis supports literal construction of arrays using the following
+        functions:
+
+        1. [`ibis.array`][ibis.array]
+        1. [`ibis.struct`][ibis.struct]
+        1. [`ibis.map`][ibis.map]
+
+        Constructing these types using `literal` will be deprecated in a future
+        release.
```

Comment:
In favor of?

Reply:
The list just above the statement. `literal` isn't going away, but support for constructing list, maps, and structs using `literal` will

Reply:
The reasons why are:

1. `literal` is starting to become a `DataFrame` constructor situation with a huge amount of branching and hard-to-follow conditions due to the more complex types
2. Adding support for things like `[a, 1, c]` is therefore not tenable
3. the lack of a 1:1 mapping between `dict` and an ibis type (since `dict` can be used as both a map and a struct) would stuff the `literal` API even more to the brim